### DISCUSSION TITLE
Help with basic troubleshooting

### DISCUSSION THREAD

**Discussion by bdudok** [pyControl/code#72] [Q&A] [answered]
Hi,
 I am looking forward to give pyControl a try! I have just got the breakout board and a few peripherals to get started. It was straightforward to follow the setup guides, and tests that only involve the board (tasks/example/all_states_example.py and blinker.by) run fine. However, I can't get any response from a rotary encoder (tasks/example/running_wheel.py) or a lickometer (tests/Board tests/lickometer_test.py). The files upload and the tasks start without error messages, but if I spin the encoder or connect the licokometer pins, nothing happens (analog stays at 0 forever, no events). Where do I get started with troubleshooting these?
 Thanks In advance!

**Comment by alustig3** [comment]
Hi,

a couple of things to check:
- Are you using the latest pyControl [version 1.8.1](https://github.com/pyControl/code/releases)?
- Are you powering the [breakout board ](https://pycontrol.readthedocs.io/en/latest/user-guide/hardware/#breakout-boards) with an external 12V power supply and not just with the USB connection?
- When you run `tasks/example/running_wheel.py`, is your [encoder](https://pycontrol.readthedocs.io/en/latest/user-guide/hardware/#rotary-encoder) connected to port1?

**Reply by bdudok** [reply]
Yes to all 3. Is 1 A adequate for the 12V?

**Reply by alustig3** [reply]
yes 1A is adequate

- What OS (Windows, Mac, or Linux) are you using? 
- if there is a framework or GUI error, an `ErrorLog.txt` file will automatically be created in the top level of your pyControl directory (alongside `pyControl_GUI.pyw` and `ChangeLog.txt` etc.). Do you have an `ErrorLog.txt` and if so, can you please share the contents?
- do you have any other peripherals besides a lickometer and rotary encoder? A nosepoke perhaps?

**Reply by bdudok** [reply]
I am using Mac. I did have some error messages before I realized that the framework and hardware  definitions have to be loaded from the GUI, but no error messages whatsoever since (not in the GUI and not in ErrorLog.txt. 
I have an audio player but no speaker yet. I don't have a nosepoke.

**Reply by alustig3** [reply]
- and just to confirm, you are using a [pyControl breakout board 1.2](https://pycontrol.readthedocs.io/en/latest/user-guide/hardware/#breakout-boards)?
- did you purchase your hardware already assembled from open-ephys/Labmaker or did you build the hardware yourself?

One unlikely possibility that could explain things is if the port_1 connector is damaged/malfunctioning. We can test this by trying the lickometer again, but this time plugged into port_2, and with an update to the `Lickometer()` instantiation so it uses port 2 like below: 

```python
# Install the driver file for the Lickometer (see
# https://pycontrol.readthedocs.io/en/latest/user-guide/hardware/#more-devices).
# Connect breakout board 1.2 to the computer, plug in the 12V power supply.  
# Connect a Lickometer to port 1 of the breakout board. Plug a houselight into
# the SOL-1 and SOL-2 ports on the Lickometer. When you make an electrical
# connection between LK1 and GND the houselight connected to SOL-1 will turn on.
# When you make an electrical connection between LK2 and GND the houselight 
# connected to SOL-2 will turn on.

from pyControl.utility import *
from devices import Breakout_1_2, Lickometer

# Define hardware

board = Breakout_1_2()
lickometer  = Lickometer(board.port_2, 
                         rising_event_A='lick_1', falling_event_A='lick_1_off',
                         rising_event_B='lick_2', falling_event_B='lick_2_off')

# State machine.

states = ['state1']

events = ['lick_1', 'lick_1_off','lick_2', 'lick_2_off']

initial_state = 'state1'

def state1(event):
    if event == 'lick_1':
        lickometer.SOL_1.on()
    elif event == 'lick_1_off':
        lickometer.SOL_1.off()
    elif event == 'lick_2':
        lickometer.SOL_2.on()
    elif event == 'lick_2_off':
        lickometer.SOL_2.off()
```

another unlikely possibility is that the problem is the ethernet cord going between the breakout board and the peripheral. The ethernet cable being the problem is unlikely, but it would be super unlikely for two ethernet cables to not work properly, so it wouldn't hurt to swap in a different ethernet cable if you have one and seeing if it produces the same failure/outcome.

@ThomasAkam any ideas?

**Comment by ThomasAkam** [comment, marked as answer]
With two different devices not working when using standard code it does make me wonder if there is a problem with the power supply such that the devices are not recieving the 5V power they need.  If everything is working, the points labled on the image below should be at 5V and 12V relative to the point labled ground (GND).  Could you check this with a multimeter? 

![image](https://user-images.githubusercontent.com/4661184/221683272-4a8c9130-0e0c-4a0a-9807-446ff8db8fd5.png)

**Reply by bdudok** [reply]
Hi, 
Yes, I purchased the 1.2 board from openEphys.
It looks like the issue may be with the 12V wallwarts I got, or a short on the board. On the pycontrol board connected to USB I have 3.3 V but only 4.47 for the 5V. On the breakout board pins you indicate, there is 0.0 for 12V (with the power supply connected and plugged in and warm) and 0.4 for the 5V, so no wonder nothing connected to that works. I have tried two power supplies but they are identical from the same box. I will try getting a different one.

**Reply by alustig3** [reply]
One thing to keep in mind is that the pyboard breakout board is expecting a barrel jack power supply in which the outside of the barrel is GND and the inside is positive voltage. You can check using a multimeter, as well as check the label on the power supply
![power_supply](https://user-images.githubusercontent.com/8128628/221709812-4a2f0a4e-2f06-4d49-a6f8-255113575db0.jpeg)

**Comment by bdudok** [comment]
Hi,
 The issue was with the power supply. Inadvertently, I initially got AC-AC (I didn't even know those exist with the same barrel connector). The test scripts work now with the proper power supply. Thanks for your help!

**Comment by ThomasAkam** [comment]
Glad to hear that fixed the problem.  I've added some information to the troubleshooting page of the docs about devices not working when the breakout board is not powered.
